runtime/src/bank.rs

```diff
@@ -12468,4 +12468,54 @@ pub(crate) mod tests {
             ))
         );
     }
+
+    #[test]
+    fn test_clean_unrooted_dropped_banks() {
```

Comment:
Note for other reviewers that there is also a special case for when the unrooted bank is also not frozen, in which case the slot pubkey diff will not be added to the dirty key list for clean. 

After some discussion, the decision was there will be an incoming fix in `Bank::drop()` to handle this case

Reply:
Yes! That fix will be in the next PR.